### PR TITLE
Rename balance to wallet

### PR DESCRIPTION
**Pull request by T-Dnzt**
Issue/Task Number: T290

# Overview

We realized the name `Balance` didn't make sense for what we were using it. This PR changes it to `Wallet`, as in a user can have many wallets and each wallet can have many balances in different currencies.

# Changes

- Rename Balance -> Wallet
- Rename ComputedBalance -> Balance
- Rename balance tables and all the foreign keys

# Implementation Details

Mostly renaming, didn't change the logic.

# Usage

I've updated/fixed the tests but haven't played much with Swagger yet. Would be awesome if you could test the calls :)

# Impact

This PR includes breaking changes for the API. The `address` object type is gone, replaced by `wallet`. All the endpoints which use to have `balance` now contain `wallet`.

### CODE REVIEW

File: apps/ewallet_api/test/ewallet_api/v1/controllers/balance_controller_test.exs [hunk 1]
```diff
@@ -1,29 +1,29 @@
-defmodule EWalletAPI.V1.ComputedBalanceControllerTest do
+defmodule EWalletAPI.V1.BalanceControllerTest do
```
Comment (T-Dnzt) [inline apps/ewallet_api/test/ewallet_api/v1/controllers/balance_controller_test.exs]: Invalid name, should be `WalletControllerTest`.

Comment (T-Dnzt) [reply apps/ewallet_api/test/ewallet_api/v1/controllers/balance_controller_test.exs]: Renamed in #197

File: apps/local_ledger_db/priv/repo/migrations/20180514095907_rename_balance_to_wallet.exs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Enum.each(tables, fn {table, columns} ->
+      Enum.each(columns, fn {_old_name, new_name} ->
+        alter table(table) do
+          modify new_name, references(:wallet, type: :string,
+                                      column: :address), null: false
+        end
+      end)
+    end)
+
+    # tables = [:cached_balance, :transaction]
```
Comment (unnawut) [inline apps/local_ledger_db/priv/repo/migrations/20180514095907_rename_balance_to_wallet.exs]: ??

File: docs/conventions.md [hunk 3]
```diff
@@ -66,32 +66,32 @@ iex> EWalletDB.User.get("9858570e-0a1f-4e4a-a630-4752aa04021c")
 %EWalletDB.User{
   # ...
   accounts: #Ecto.Association.NotLoaded<association :accounts is not loaded>,
-  balances: #Ecto.Association.NotLoaded<association :users is not loaded>,
+  wallets: #Ecto.Association.NotLoaded<association :users is not loaded>,
```
Comment (unnawut) [inline docs/conventions.md]: Not caused by this PR but...

`association :users is not loaded` -> `association :wallets is not loaded`

File: docs/design/entities.md [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+- __Accounts__: A grouping of users and wallets. It can be used by a provider admin to define logical groupings according to its business needs, e.g. an account for each product branding, each store branch, each sales team or each operating region, etc.
 
 - __Users__: A user can either be a regular user or a staff user (a.k.a admin) In the first case, it would be someone using the end-user applications created by providers (mobile applications for example), who was added to the eWallet database by the provider. In the second case, it's a user able to access the OmiseGO admin panel.
 
-- __Balances__: A balance identified by an address which is used to group transactions and holds a value. A user can have more than one balance if needed, and a balance can contain different currencies. You can learn more about balances [there](/docs/balances.md).
+- __Balances__: A balance identified by an address which is used to group transactions and holds a value. A user can have more than one balance if needed, and a balance can contain different currencies. You can learn more about wallets [there](/docs/wallets.md).
```
Comment (unnawut) [inline docs/design/entities.md]: Also change the bold text "Balances" to "Wallets"?

File: docs/setup/env.md [hunk 5]
```diff
@@ -51,7 +51,7 @@ Examples:
 
 If this feature is enabled, you can also specify a caching strategy.
 
-- `BALANCE_CACHING_STRATEGY`: Specify if new cached balances should be computed using a previous cache or by recalculating everything from scratch.
+- `BALANCE_CACHING_STRATEGY`: Specify if new cached wallets should be computed using a previous cache or by recalculating everything from scratch.
```
Comment (unnawut) [inline docs/setup/env.md]: Is it "cached wallets" or should remain "cached balances" still...

File: docs/setup/integration.md [hunk 6]
```diff
@@ -9,9 +9,9 @@ Integrating the OmiseGO eWallet requires a new setup to be deployed. Feel free t
 |Area of responsibilities|Tasks|
 |------------------------|-----|
 |Token management   | - Create loyalty tokens <br> - Put more loyalty tokens in circulation <br> - Remove loyalty tokens from circulation <br> - Provide user interface for creating new loyalty tokens <br> - Provide user interface for add/remove of loyalty tokens from circulation|
-|Secondary user store|- Create users along with their token balances. The user stored in the Wallet API is solely for identifying and transacting with the user’s token balances.|
+|Secondary user store|- Create users along with their token wallets. The user stored in the Wallet API is solely for identifying and transacting with the user’s token wallets.|
```
Comment (unnawut) [inline docs/setup/integration.md]: Do we still need the word "token" before wallets? It's not used anywhere else and still understandable I think.

**Pull request merged**